src/components/timeline/TrackScreenshots.js

```diff
@@ -0,0 +1,250 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import { createPortal } from 'react-dom';
+import explicitConnect from '../../utils/connect';
+import {
+  selectorsForThread,
+  getCommittedRange,
+  getPreviewSelection,
+} from '../../reducers/profile-view';
+import { withSize, type SizeProps } from '../shared/WithSize';
+
+import type { ThreadIndex, Thread } from '../../types/profile';
+import type { ScreenshotPayload } from '../../types/markers';
+import type { TracingMarker } from '../../types/profile-derived';
+import type { Milliseconds } from '../../types/units';
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+
+import { ensureExists } from '../../utils/flow';
+import './TrackScreenshots.css';
+
+type OwnProps = {|
+  +threadIndex: ThreadIndex,
+  +screenshotId: string,
```

Comment:
Not to be changed in this PR:

Sometimes this value is called `screenshotId` and sometimes `windowID`; it's a bit confusing to read and especially to follow along across the code base. If you have a good idea to rename this consistently I'd be happy (in another PR).

Reply:
`windowID` is from the marker data, but I'll name all of our code to use `windowId`, to be similar, but also tweak the capitalization to be in-line with our style.